For bootloader build, it has an error that says
`../../../source/hic_hal/stm32/stm32h743ii/extended_features/source/DAP_vendor_ex.c:1:10: fatal error: DAP.h: No such file or directory`

--- a/projects.yaml
+++ b/projects.yaml
@@ -183,6 +183,7 @@ projects:
         - *module_if
         - *module_hic_stm32h743ii
         - records/family/all_family.yaml
+        - records/board/stm32h743ii_if.yaml
         - records/usb/usb-bulk.yaml
     sam3u2c_mkit_dk_dongle_nrf5x_if:
         - *module_if

--- a/records/board/stm32h743ii_if.yaml
+++ b/records/board/stm32h743ii_if.yaml
@@ -1,6 +1,10 @@
 common:
+    includes:
+        - source/hic_hal/stm32/stm32h743ii/extended_features/include
     sources:
         board:
             - source/board/stm32h743ii.c
         family:
             - source/family/st/stm32h743ii/target.c
+        hic_hal:
+            - source/hic_hal/stm32/stm32h743ii/extended_features/source

--- a/records/hic_hal/stm32h743ii.yaml
+++ b/records/hic_hal/stm32h743ii.yaml
@@ -18,14 +18,12 @@ common:
         - source/hic_hal/stm32/stm32h743ii/STM32H7xx_HAL_Driver
         - source/hic_hal/stm32/stm32h743ii/STM32H7xx_HAL_Driver/Inc
         - source/hic_hal/stm32/stm32h743ii/STCubeGenerated/Inc
-        - source/hic_hal/stm32/stm32h743ii/extended_features/include
     sources:
         hic_hal:
             - source/hic_hal/stm32/stm32h743ii
             - source/hic_hal/stm32/stm32h743ii/cmsis
             - source/hic_hal/stm32/stm32h743ii/STM32H7xx_HAL_Driver/Src
             - source/hic_hal/stm32/stm32h743ii/STCubeGenerated/Src
-            - source/hic_hal/stm32/stm32h743ii/extended_features/source
 tool_specific:
     uvision:
         misc:

--- a/source/board/stm32h743ii.c
+++ b/source/board/stm32h743ii.c
@@ -21,6 +21,7 @@
 
 #include "target_family.h"
 #include "target_board.h"
+#include "i2c.h"
 
 static void prerun_board_config(void) {
     I2C_DAP_Initialize();
@@ -32,5 +33,5 @@ const board_info_t g_board_info = {
     .family_id = kStub_HWReset_FamilyID,
     .daplink_drive_name = "DAPLINK",
     .prerun_board_config = prerun_board_config,
-    .target_cfg = &target_device,
+    // .target_cfg = &target_device,   This board doesn't have a target
 };

--- a/source/daplink/cmsis-dap/DAP_vendor.c
+++ b/source/daplink/cmsis-dap/DAP_vendor.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2013-2016 ARM Limited. All rights reserved.
+ * Copyright (c) 2013-2020 Arm Limited. All rights reserved.
  * Copyright 2019, Cypress Semiconductor Corporation
  * or a subsidiary of Cypress Semiconductor Corporation.
  *
@@ -31,7 +31,8 @@
 #include "DAP_config.h"
 #include "DAP.h"
 #include "info.h"
-#include "main.h"
+#include "daplink.h"
+#include DAPLINK_MAIN_HEADER
 #include "uart.h"
 #include "settings.h"
 #include "target_family.h"

--- a/source/family/st/stm32h743ii/flash_blob.c
+++ b/source/family/st/stm32h743ii/flash_blob.c
@@ -1,0 +1,2 @@
+// Placeholder, use the tools/flash_algo.py tp generate the appropriate flash algo blob
+static const program_target_t flash = {0};

--- a/source/family/st/stm32h743ii/target.c
+++ b/source/family/st/stm32h743ii/target.c
@@ -1,0 +1,50 @@
+/**
+ * @file    target.c
+ * @brief   Target information for the
+ *
+ * DAPLink Interface Firmware
+ * Copyright (c) 2017-2019, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "target_config.h"
+#include "daplink_addr.h"
+
+// The file flash_blob.c must only be included in target.c
+#include "flash_blob.c"
+
+/**
+* List of start and size for each size of flash sector
+* The size will apply to all sectors between the listed address and the next address
+* in the list.
+* The last pair in the list will have sectors starting at that address and ending
+* at address start + size.
+*/
+static const sector_info_t sectors_info[] = {
+    {DAPLINK_ROM_IF_START, DAPLINK_SECTOR_SIZE},
+ };
+
+
+target_cfg_t target_device = {
+    .sectors_info               = sectors_info,
+    .sector_info_length         = (sizeof(sectors_info))/(sizeof(sector_info_t)),
+    .flash_regions[0].start     = DAPLINK_ROM_START,
+    .flash_regions[0].end       = DAPLINK_ROM_START + DAPLINK_ROM_SIZE,
+    .flash_regions[0].flags     = kRegionIsDefault,
+    .flash_regions[0].flash_algo= (program_target_t *) &flash,
+    .ram_regions[0].start       = DAPLINK_RAM_APP_START,
+    .ram_regions[0].end         = DAPLINK_RAM_APP_START + DAPLINK_RAM_APP_SIZE,
+};
+

